lib/LANraragi/Model/Search.pm

```diff
@@ -370,38 +370,22 @@ sub compute_search_filter {
 sub sort_results {
 
     my ( $sortkey, $sortorder, @filtered ) = @_;
-
     my $redis = LANraragi::Model::Config->get_redis;
-
-    @filtered = sort {
-
-        my $tags_a = $redis->hget( $a, "tags" );
-        my $tags_b = $redis->hget( $b, "tags" );
-
-        # Not a very good way to make items end at the bottom...
-        my $meta1 = "zzzz";
-        my $meta2 = "zzzz";
-
-        if ( $sortkey ne "title" ) {
-            my $re = qr/$sortkey/;
-            if ( $tags_a =~ m/.*${re}:(.*)(\,.*|$)/ ) {
-                $meta1 = $1;
-            }
-
-            if ( $tags_b =~ m/.*${re}:(.*)(\,.*|$)/ ) {
-                $meta2 = $1;
-            }
-        }
-
-        if ($sortorder) {
-            ncmp( lc($meta2), lc($meta1) );
-        } else {
-            ncmp( lc($meta1), lc($meta2) );
-        }
-
-    } @filtered;
-
-    return @filtered;
+	
+    my $re = qr/$sortkey/;
+	
+	my %tmpfilter = map {$_ => ($redis->hget( $_, "tags" ) =~ m/.*${re}:(.*)(\,.*|$)/) ? $1 : "zzzz" } @filtered;
+	
+	my @sorted = map { $_->[0] }
+	  sort { ncmp($a->[1], $b->[1]) }
+	  map  { [$_, lc($tmpfilter{$_})] }
+	  keys %tmpfilter;
```

Comment:
```suggestion
	my @sorted = map { $_->[0] } # Map back to only having the ID
	  sort { ncmp($a->[1], $b->[1]) } # Sort by the tag
	  map  { [$_, lc($tmpfilter{$_})] } # Map to an array containing the ID and the lowercased tag
	  keys %tmpfilter; # List of IDs
```

I guess you could replace `keys %tmpfilter` by `@filtered` here? It doesn't really change much I guess..

Reply:
The hash value contains {ID: TAG}.
Therefore, this code commonly used hash called %tmpfilter to put the id value sorted based on tag.
I don't know if @filtered will work properly.

Reply:
Calling multiple variables will result in performance loss, so it seems better to have the same %tmpfilter called.